packages/server/src/Connection.ts

```diff
@@ -216,21 +220,40 @@ export class Connection {
    * @public
    */
   public handleMessage(data: Uint8Array): void {
+    if (this.beforeHandleMessageQueue.length > 0 || this.beforeHandleMessageProcessing) {
+      this.beforeHandleMessageQueue.push(data)
+      return
+    }
+
+    this.processBeforeHandleMessage(data)
+  }
+
+  /** Process an incoming message */
+  private processBeforeHandleMessage(data: Uint8Array): void {
```

Comment:
I would prefer using async-await syntax for this method. This would allow you to use a `while (this. beforeHandleMessageQueue.length > 0` loop.